src/lib/tls/tls13/tls_channel_impl_13.cpp

```diff
@@ -105,6 +107,15 @@ size_t Channel_Impl_13::received_data(const uint8_t input[], size_t input_size)
                {
                while(auto handshake_msg = m_handshake_layer.next_message(policy(), m_transcript_hash))
                   {
+                  // After the initial handshake message is received, the record
+                  // layer must be more restrictive.
+                  // See RFC 8446 5.1 regarding "legacy_record_version"
+                  if(m_first_message_received)
+                     {
+                     m_record_layer.disable_receiving_compat_mode();
+                     m_first_message_received = true;
+                     }
```

Comment:
Multiple issues here, maybe all caused by some wrong commit:
- Does not make sense to set `m_first_message_received` to true in the branch there it is true.
- I don't see `m_first_message_received` being set to true anywhere else.
- If the condition will get true, how to avoid setting `disable_receiving_compat_mode()` again in each loop iteration?

Reply:
Oi! What a blunder. Turns out the `if()` condition needed to be negated. I independently stumbled over this while debugging some server BoGo test.